core/src/main/scala/org/apache/spark/storage/DiskBlockManager.scala

```diff
@@ -100,7 +100,16 @@ private[spark] class DiskBlockManager(conf: SparkConf, deleteFilesOnStop: Boolea
 
   /** List all the blocks currently stored on disk by the disk manager. */
   def getAllBlocks(): Seq[BlockId] = {
-    getAllFiles().map(f => BlockId(f.getName))
+    getAllFiles().flatMap { f =>
+      val blockId = BlockId.guess(f.getName)
```

Comment:
It looks not so necessary to define a new method `guess` for the use here only. I think here we can still use `apply` and catch/log the exception. In another word, we can simply changes `apply()` and use it here, defining new `guess` method is not so necessary.

Reply:
+1 for using try-catch here.

Reply:
Will do. Should I log the exception even if the file has been produced by `Utils.tempFileWith`?

Reply:
I think we don't need to log here.

Reply:
Do you mean we don't need to log at all?